openpype/hosts/fusion/addon.py

```diff
@@ -60,8 +60,18 @@ def get_launch_hook_paths(self, app):
             return []
         return [os.path.join(FUSION_HOST_DIR, "hooks")]
 
-    def add_implementation_envs(self, env, _app):
+    def add_implementation_envs(self, env, app):
         # Set default values if are not already set via settings
+        app_version = app.name
+        if int(app_version) < 18:
```

Comment:
Be aware that app names can be customized (they can literally be any string). That discussion also came up when implementing [this](https://github.com/ynput/OpenPype/blob/00b83e58792db4e33d5d218ce942a8667d724265/openpype/hosts/fusion/addon.py#L20-L47) in this PR: https://github.com/ynput/OpenPype/pull/4572

If anyone had a `18.5.beta` version registered this would currently just error on it.

Additionally, Fusion 18 can also still use Python 3.6. With Fusion 18 I believe they only added the possibility to a Python between 3.3 and 3.10 as Python 3 interpreter. If so, this issue could be just as relevant for Fusion 18+

Reply:
Now used `get_fusion_version`.

What would be the safer option? Trying to import `requests` in `openpype/hosts/fusion/deploy/MenuScripts/openpype_menu.py` and modify environment only if it fails there? That doesn't look particularly elegant.

Reply:
The best solution would be packaging the right tools for the right python interpreters - e.g. using Rez #4641 

A hacky solution could indeed be hot-patching it within the interpreter, but it'd only fix the Python session if we were to run OpenPype tools - but other Py3 scripts Fusion would then run that might use a similar dependency would then still break. So it's not as nice as making sure we're not adding the wrong packages to Fusion's Python.